### PR TITLE
fix(CDAP-20559): Use system context while fetching the SCM token

### PR DESCRIPTION
**Pull request by de-lan**
## What

Use system context while fetching the SCM token. Set back to user principal after the fetch

### CODE REVIEW

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java [hunk 1]
```diff
@@ -76,18 +78,33 @@ private static class SecureStorePasswordProvider extends RefreshableCredentialsP
...
       try {
         data = secureStore.getData(namespaceId, passwordKeyName);
+
+        if (data == null) {
+          throw new AuthenticationConfigException(
+              String.format("Password with key name %s not found in secure store",
+                  passwordKeyName));
+        }
+        
+        password = new String(data, StandardCharsets.UTF_8);
```
Comment (dli357) [inline cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java:97]: nit+optional: Maybe we should change `password` to `personalAccessToken` everywhere to properly reflect what is being fetched?

**Review by dli357 — APPROVED** [review]
LGTM, thanks!

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java (hunk 1 above)
Comment (de-lan) [reply cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java:97]: yeah I agree. But seems like we made it `password` to reflect the fact that we're using `UsernamePasswordCredentialsProvider` in jgit (it accepts access token or username+password). Correct me if I'm wrong fyi @arjan-bal

Comment (arjan-bal) [reply cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java:97]: That's correct. This inner class `SecureStorePasswordProvider ` can be re-used with Username password auth later. It doesn't do anything specific to PAT.

Comment (de-lan) [reply cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPatAuthenticationStrategy.java:97]: @arjan-bal Thanks for the confirm, merging it now

**Pull request merged**